? applyclose.diff
? dd-filter.diff
? debug
? foo.dat
? mac-build
? x11-debug
Index: docs/latex/wx/dirctrl.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/dirctrl.tex,v
retrieving revision 1.10
diff -u -u -2 -r1.10 dirctrl.tex
--- docs/latex/wx/dirctrl.tex	2006/12/24 12:38:38	1.10
+++ docs/latex/wx/dirctrl.tex	2007/05/15 17:25:17
@@ -24,5 +24,4 @@
 \twocolitem{\indexit{wxDIRCTRL\_3D\_INTERNAL}}{Use 3D borders for internal controls.}
 \twocolitem{\indexit{wxDIRCTRL\_SELECT\_FIRST}}{When setting the default path, select the first file in the directory.}
-\twocolitem{\indexit{wxDIRCTRL\_SHOW\_FILTERS}}{Show the drop-down filter list.}
 \twocolitem{\indexit{wxDIRCTRL\_EDIT\_LABELS}}{Allow the folder and file labels to be editable.}
 \end{twocollist}
Index: include/wx/generic/dirctrlg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/dirctrlg.h,v
retrieving revision 1.34
diff -u -u -2 -r1.34 dirctrlg.h
--- include/wx/generic/dirctrlg.h	2007/03/07 11:34:29	1.34
+++ include/wx/generic/dirctrlg.h	2007/05/15 17:25:17
@@ -45,5 +45,5 @@
     // When setting the default path, select the first file in the directory
     wxDIRCTRL_SELECT_FIRST   = 0x0020,
-    // Show the filter list
+    // unused, for compatibility only
     wxDIRCTRL_SHOW_FILTERS   = 0x0040,
     // Use 3D borders on internal controls
@@ -185,4 +185,5 @@
     wxString        m_defaultPath; // Starting path
     long            m_styleEx; // Extended style
+    long            m_filterStyle; // Style of m_filterListCtrl
     wxString        m_filter;  // Wildcards in same format as per wxFileDialog
     int             m_currentFilter; // The current filter index
Index: samples/widgets/dirctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/widgets/dirctrl.cpp,v
retrieving revision 1.3
diff -u -u -2 -r1.3 dirctrl.cpp
--- samples/widgets/dirctrl.cpp	2006/12/03 13:43:42	1.3
+++ samples/widgets/dirctrl.cpp	2007/05/15 17:25:18
@@ -133,6 +133,7 @@
                *m_chk3D,
                *m_chkFirst,
-               *m_chkFilters,
                *m_chkLabels;
+    // filters
+    wxCheckBox *m_fltr[3];
 
 private:
@@ -183,8 +184,15 @@
     m_chk3D      = CreateCheckBoxAndAddToSizer(sizerUseFlags, _T("wxDIRCTRL_3D_INTERNAL"));
     m_chkFirst   = CreateCheckBoxAndAddToSizer(sizerUseFlags, _T("wxDIRCTRL_SELECT_FIRST"));
-    m_chkFilters = CreateCheckBoxAndAddToSizer(sizerUseFlags, _T("wxDIRCTRL_SHOW_FILTERS"));
     m_chkLabels  = CreateCheckBoxAndAddToSizer(sizerUseFlags, _T("wxDIRCTRL_EDIT_LABELS"));
     sizerLeft->Add(sizerUseFlags, wxSizerFlags().Expand().Border());
 
+    wxSizer *sizerFilters =
+        new wxStaticBoxSizer(wxVERTICAL, this, _T("&Filters"));
+    m_fltr[0] = CreateCheckBoxAndAddToSizer(sizerFilters, wxString::Format(wxT("all files (%s)|%s"),
+                            wxFileSelectorDefaultWildcardStr, wxFileSelectorDefaultWildcardStr));
+    m_fltr[1] = CreateCheckBoxAndAddToSizer(sizerFilters, wxT("C++ files (*.cpp; *.h)|*.cpp;*.h"));
+    m_fltr[2] = CreateCheckBoxAndAddToSizer(sizerFilters, wxT("PNG images (*.png)|*.png"));
+    sizerLeft->Add(sizerFilters, wxSizerFlags().Expand().Border());
+
     wxButton *btn = new wxButton(this, DirCtrlPage_Reset, _T("&Reset"));
     sizerLeft->Add(btn, 0, wxALIGN_CENTRE_HORIZONTAL | wxALL, 15);
@@ -237,7 +245,16 @@
         ( m_chk3D->IsChecked() ? wxDIRCTRL_3D_INTERNAL : 0 ) |
         ( m_chkFirst->IsChecked() ? wxDIRCTRL_SELECT_FIRST : 0 ) |
-        ( m_chkFilters->IsChecked() ? wxDIRCTRL_SHOW_FILTERS : 0 ) |
         ( m_chkLabels->IsChecked() ? wxDIRCTRL_EDIT_LABELS : 0 )
     );
+
+    wxString filter;
+    for (int i = 0; i < 3; ++i) 
+        if (m_fltr[i]->IsChecked()) 
+        {
+            if (!filter.IsEmpty())
+                filter += wxT("|");
+            filter += m_fltr[i]->GetLabel();
+        }
+    dirCtrl->SetFilter(filter);
 
     // update sizer's child window
Index: src/generic/dirctrlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/dirctrlg.cpp,v
retrieving revision 1.166
diff -u -u -2 -r1.166 dirctrlg.cpp
--- src/generic/dirctrlg.cpp	2007/04/11 00:22:58	1.166
+++ src/generic/dirctrlg.cpp	2007/05/15 17:25:18
@@ -471,5 +471,4 @@
     wxFLAGS_MEMBER(wxDIRCTRL_3D_INTERNAL)
     wxFLAGS_MEMBER(wxDIRCTRL_SELECT_FIRST)
-    wxFLAGS_MEMBER(wxDIRCTRL_SHOW_FILTERS)
 
 wxEND_FLAGS( wxGenericDirCtrlStyle )
@@ -565,15 +564,15 @@
         treeStyle |= wxBORDER_SUNKEN;
 
-    long filterStyle = 0;
+    m_filterStyle = 0;
     if ((style & wxDIRCTRL_3D_INTERNAL) == 0)
-        filterStyle |= wxNO_BORDER;
+        m_filterStyle |= wxNO_BORDER;
     else
-        filterStyle |= wxBORDER_SUNKEN;
+        m_filterStyle |= wxBORDER_SUNKEN;
 
     m_treeCtrl = CreateTreeCtrl(this, wxID_TREECTRL,
                                 wxPoint(0,0), GetClientSize(), treeStyle);
 
-    if (!filter.empty() && (style & wxDIRCTRL_SHOW_FILTERS))
-        m_filterListCtrl = new wxDirFilterListCtrl(this, wxID_FILTERLISTCTRL, wxDefaultPosition, wxDefaultSize, filterStyle);
+    if (!filter.empty())
+        m_filterListCtrl = new wxDirFilterListCtrl(this, wxID_FILTERLISTCTRL, wxDefaultPosition, wxDefaultSize, m_filterStyle);
 
     m_defaultPath = dir;
@@ -1183,4 +1182,11 @@
     m_filter = filter;
 
+    if (!filter.empty() && !m_filterListCtrl)
+        m_filterListCtrl = new wxDirFilterListCtrl(this, wxID_FILTERLISTCTRL, wxDefaultPosition, wxDefaultSize, m_filterStyle);
+    else if (filter.empty() && m_filterListCtrl) {
+        m_filterListCtrl->Destroy();
+        m_filterListCtrl = NULL;
+    }
+
     wxString f, d;
     if (ExtractWildcard(m_filter, m_currentFilter, f, d))
@@ -1192,4 +1198,8 @@
         m_currentFilterStr = wxT("*.*");
 #endif
+    // current filter index is meaningless after filter change, set it to zero
+    SetFilterIndex(0);
+    if (m_filterListCtrl)
+        m_filterListCtrl->FillFilterList(m_filter, 0);
 }
 
Index: src/xrc/xh_gdctl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xrc/xh_gdctl.cpp,v
retrieving revision 1.11
diff -u -u -2 -r1.11 xh_gdctl.cpp
--- src/xrc/xh_gdctl.cpp	2007/03/10 17:22:51	1.11
+++ src/xrc/xh_gdctl.cpp	2007/05/15 17:25:19
@@ -34,5 +34,4 @@
     XRC_ADD_STYLE(wxDIRCTRL_3D_INTERNAL);
     XRC_ADD_STYLE(wxDIRCTRL_SELECT_FIRST);
-    XRC_ADD_STYLE(wxDIRCTRL_SHOW_FILTERS);
     XRC_ADD_STYLE(wxDIRCTRL_EDIT_LABELS);
     AddWindowStyles();

 	  	 
